### DISCUSSION TITLE
How to do cable trace in PyNetbox

### DISCUSSION THREAD

**Discussion by Havekk** [netbox-community/netbox#6411] [Q&A] [answered]
Hi.  I'm new to Netbox / programming / Python but managed to work out most bits, but I cant work out how to do a cable trace with pynetbox.

The API lists the following but not sure how to call that:
![image](https://user-images.githubusercontent.com/84136312/118134925-5984bf80-b3fa-11eb-8afd-ecbad076f0b1.png)

I have tried the following:
Trace = netbox.circuits.circuit_terminations.trace(id=TermID)
Trace = netbox.circuits.circuit_terminations.trace.get(id=TermID)

But get:
AttributeError: 'Endpoint' object has no attribute 'trace'

I'm sure its obvious, but not worked it out from docs and examples.  I'm looking for the easiest way to determine the device and interface a circuit is connected to.

Very impressed with netbox so far!

Thanks.

**Comment by candlerb** [comment, marked as answer]
It's probably an oversight in pynetbox.  This works (for an interface):

```
>>> z = nb.dcim.interfaces.get(2)
>>> z.trace()
[[enp3s0, Cable #48, ether3]]
```

But this doesn't (for a circuit termination):

```
>>> ct = nb.circuits.circuit_terminations.get(1)
>>> ct.trace()
...
AttributeError: object has no attribute "trace"
```

**Comment by dionrowney** [comment]
Also note that this seems to not work for front-ports or rear-ports also.  Even though one can view a GUI trace anywhere alone the way.  One can see that /dcim/front-ports/12412/trace/ works but /api/dcim/front-ports/12412/trace/ does not.  Only works with device interfaces via api.

**Reply by CADbloke** [reply]
for future visitors, front & rear ports use `/paths` instead of `/trace`
